### PR TITLE
test: add test ids to watch components

### PR DESCRIPTION
**Pull request by DanChae03**
# Description

Add test ids to watch components. Skipped over the tedious ones for now. Also corrected test id naming convention to UpperCase and fixed the corresponding unit tests.

Components without test id added: 

Non-interactable:
HeroOverlay

No place to put data test id:
VideoPlayer (But videocontrols has one on same level)

### <samp>🤖 Generated by Copilot at 8aaff54</samp>

This pull request adds or updates the `data-testid` attributes of various app components for testing with the `@testing-library/react` library. It also modifies some tests to use the `getByTestId` query instead of the `getByLabelText` query. The purpose of these changes is to improve the testability and consistency of the app components.

- Link to Basecamp Todo

# How should this PR be QA Tested?

Please describe the QA tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

# Walkthrough

### <samp>🤖 Generated by Copilot at 8aaff54</samp>

*  Add `dataTestId` props and `data-testid` attributes to various components to enable testing with `@testing-library/react` ([link](https://github.com/JesusFilm/core/pull/1855/files?diff=unified&w=0#diff-a346101824fb43fb02421e273725ab59e59afc964e32103e416da450f3d158d0R124), [link](https://github.com/JesusFilm/core/pull/1855/files?diff=unified&w=0#diff-5845364693724837babd6740757fb714fc150953a2240fb57ccbde3dcc1e9930R82), [link](https://github.com/JesusFilm/core/pull/1855/files?diff=unified&w=0#diff-73a7268c2c3cf67a0936aafbc1c4126754f9779fa60f2353412c9cc2bbb96bc4R20), [link](https://github.com/JesusFilm/core/pull/1855/files?diff=unified&w=0#diff-8ab397ee0fd015c198546b901332cf8844844ee44d92e89d9c79a5c05c967680L8-R8), [link](https://github.com/JesusFilm/core/pull/1855/files?diff=unified&w=0#diff-b5007b71342c15ebdd6fe008c745c50040df924839c49cb2eed793ae33bf1405R30), [link](https://github.com/JesusFilm/core/pull/1855/files?diff=unified&w=0#diff-095c84dbd68df431c55ee2103e90f44421591fb7296b6ecf4bc6eae7433c98f8R107), [link](https://github.com/JesusFilm/core/pull/1855/files?diff=unified&w=0#diff-606250f961d585116f134e03e95fddfaf2d46687821cc2e3d0bb44bb4b646284R116), [link](https://github.com/JesusFilm/core/pull/1855/files?diff=unified&w=0#diff-a504696bd23c7b295553ac6518396a3d861c14de1e5e0d3d08c4ac2a5bacc74bR23), [link](https://github.com/JesusFilm/core/pull/1855/files?diff=unified&w=0#diff-dc6c78ab9451c9ba5070661f45ff6af2aa08a5003ba25751d268d23f9dd5b5c4R31), [link](https://github.com/JesusFilm/core/pull/1855/files?diff=unified&w=0#diff-d13cb57964e6a7a261d72569fbfd7d4e6467d5426de8193f1cc7dba103aa0378R54), [link](https://github.com/JesusFilm/core/pull/1855/files?diff=unified&w=0#diff-8b4953c744e40bcf3cb022a60f618632397e7a9d25e89cf7750f7cb9ff61d41fL28-R28), [link](https://github.com/JesusFilm/core/pull/1855/files?diff=unified&w=0#diff-f392dfcaa116c97c47e7e2ad44cf2627e82b00e667233ec1684a102c6cc87ebdR82), [link](https://github.com/JesusFilm/core/pull/1855/files?diff=unified&w=0#diff-56b63b1c3d3478757ed55ac1f46774878955d790bba3dd024b04ba1cb38e725cR42), [link](https://github.com/JesusFilm/core/pull/1855/files?diff=unified&w=0#diff-7f398d66d0f69d84724bdcffda2248f3826a94993930bb6b4e0b26ed772ce8ebR309), [link](https://github.com/JesusFilm/core/pull/1855/files?diff=unified&w=0#diff-42b81271f33bff2e3469a37294bf3148837541ecad948cfc4e4219543f32ef60L189-R189), [link](https://github.com/JesusFilm/core/pull/1855/files?diff=unified&w=0#diff-87dad2ed6bf79789411847f7c8a62fdb067a6a82c17d3a925972ae76661f8506R34))
* Update `data-testid` attribute values to use a more consistent and descriptive naming convention for some components ([link](https://github.com/JesusFilm/core/pull/1855/files?diff=unified&w=0#diff-1cf708f9f38f9a04099f26dc9a0a76682d23c65e6248e613a3c9e39de70c0b07L22-R25), [link](https://github.com/JesusFilm/core/pull/1855/files?diff=unified&w=0#diff-9ef5a0e9cf35d865f0bf66e5db640615cc555f8ad4d50664870c21d51e5c7afdL22-R25), [link](https://github.com/JesusFilm/core/pull/1855/files?diff=unified&w=0#diff-93adcf31eb9a43d41205e269e7e3ddc9df498de674852ade011f09c34d104a06L32-R32))
* Remove `aria-label` attribute from `SeeAllVideos` component to avoid redundancy or conflict with `data-testid` attribute ([link](https://github.com/JesusFilm/core/pull/1855/files?diff=unified&w=0#diff-8ab397ee0fd015c198546b901332cf8844844ee44d92e89d9c79a5c05c967680L14-R18))
* Modify tests for `SeeAllVideos` component to use `getByTestId` query instead of `getByLabelText` query and to match the updated `data-testid` attribute value ([link](https://github.com/JesusFilm/core/pull/1855/files?diff=unified&w=0#diff-ea4f4afb22693eeb24304c1e5b1186750e45054577dc6267c96aed6000c77d68L7-R8))

### CODE REVIEW

File: apps/watch/src/components/HomePage/SeeAllVideos/SeeAllVideos.tsx [hunk 1]
```diff
@@ -5,17 +5,17 @@ import { ReactElement } from 'react'
...
         <Button
           size="small"
           color="secondary"
           variant="outlined"
-          aria-label="all-videos-button"
           sx={{
             width: '15%',
             whiteSpace: 'nowrap'
           }}
+          data-testid="AllVideosButton"
```
Comment (csiyang) [inline apps/watch/src/components/HomePage/SeeAllVideos/SeeAllVideos.tsx:18]:
```suggestion
          data-testid="SeeAllVideos"
```

File: apps/watch/src/components/PageWrapper/PageWrapper.tsx [hunk 2]
```diff
@@ -27,6 +27,7 @@ export function PageWrapper({
       <Stack
         justifyContent="space-between"
         sx={{ width: '100%', height: '100%' }}
+        data-testid="PageWrapper"
```
Comment (csiyang) [inline apps/watch/src/components/PageWrapper/PageWrapper.tsx:30]: This component should have its testId passed in like the `Dialog` component

**Review by csiyang — CHANGES_REQUESTED** [review]

**Comment by csiyang** [comment]
Some more changes here https://github.com/JesusFilm/core/pull/1856

File: apps/watch/src/components/HomePage/SeeAllVideos/SeeAllVideos.tsx [hunk 3]
```diff
@@ -11,11 +11,11 @@ export function SeeAllVideos(): ReactElement {
           size="small"
           color="secondary"
           variant="outlined"
-          aria-label="all-videos-button"
```
Comment (kiran-redhat) [inline apps/watch/src/components/HomePage/SeeAllVideos/SeeAllVideos.tsx:14]: Do we not need this anymore ?

File: apps/watch/src/components/ShareDialog/ShareDialog.tsx [hunk 4]
```diff
@@ -104,6 +104,7 @@ export function ShareDialog({
         closeButton: true
       }}
       divider
+      testId="ShareDialog"
```
Comment (kiran-redhat) [inline apps/watch/src/components/ShareDialog/ShareDialog.tsx:107]: Will it later get injected into parewrapper with 'data-testid' ?

**Review by kiran-redhat — COMMENTED** [review]
Hey @DanChae03 , looks great. Left 2 quick questions

File: apps/watch/src/components/ShareDialog/ShareDialog.tsx (hunk 4 above)
Comment (csiyang) [reply apps/watch/src/components/ShareDialog/ShareDialog.tsx:107]: Its being passed to the shared `Dialog` component

File: apps/watch/src/components/HomePage/SeeAllVideos/SeeAllVideos.tsx (hunk 3 above)
Comment (csiyang) [reply apps/watch/src/components/HomePage/SeeAllVideos/SeeAllVideos.tsx:14]: Not technically, but we can keep it if you still need it?

**Review by csiyang — COMMENTED** [review]
@kiran-redhat Just responded to your comments since Dan won't be on again until next week. I've also reviewed the code already and it looks fine to me too. 

Are you happy with the testid's and where there are placed?

File: apps/watch/src/components/HomePage/SeeAllVideos/SeeAllVideos.tsx (hunk 3 above)
Comment (kiran-redhat) [reply apps/watch/src/components/HomePage/SeeAllVideos/SeeAllVideos.tsx:14]: Was checking if it was removed unintentionally - all good.

**Review by kiran-redhat — APPROVED** [review]
Looks sweet, thanks.

**Review by csiyang — APPROVED** [review]

**Pull request merged**